plugins/catalog/src/components/ResultsFilter/ResultsFilterNew.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Button,
+  Checkbox,
+  Divider,
+  List,
+  ListItem,
+  ListItemText,
+  makeStyles,
+  Theme,
+  Typography,
+} from '@material-ui/core';
+import React, { useCallback } from 'react';
+import { useEntityListState, applyFilter, removeFilter } from '../../state';
+
+const useStyles = makeStyles<Theme>(theme => ({
+  filterBox: {
+    display: 'flex',
+    margin: theme.spacing(2, 0, 0, 0),
+  },
+  filterBoxTitle: {
+    margin: theme.spacing(1, 0, 0, 1),
+    fontWeight: 'bold',
+    flex: 1,
+  },
+  title: {
+    margin: theme.spacing(1, 0, 0, 1),
+    textTransform: 'uppercase',
+    fontSize: 12,
+    fontWeight: 'bold',
+  },
+  checkbox: {
+    padding: theme.spacing(0, 1, 0, 1),
+  },
+}));
+
+type Props = {
+  availableTags: string[];
+};
+
+/**
+ * The additional results filter in the sidebar.
+ */
+export const ResultsFilterNew = ({ availableTags }: Props) => {
+  const classes = useStyles();
+  const { state, dispatch } = useEntityListState();
```

Comment:
One more thing: The api here could be `const { setFilter, clearFilter } = useFilterState('metadata.tags')` so we'd hide all the underlying implementation from the users

Reply:
likey 👍

Reply:
it's a bit harder to make that though :) because the dispatch fn now needs to curry in the id but must not be regenerated over and over again :)